applications/asset_tracker_v2/src/modules/Kconfig.cloud_module

```diff
@@ -20,6 +20,16 @@ config NRF_CLOUD_AGPS
 	depends on GNSS_MODULE
 	default y
 
+if NRF_CLOUD_PGPS && !NRF_CLOUD_PGPS_PARTITION
+
+config CLOUD_PGPS_MCUBOOT_PARTITION
```

Comment:
Maybe there is a different way to do what I wanted to do.

This section ensures that a secondary mcuboot partition exists for P-GPS to store its data in, since the user has disabled the P-GPS partition.

The justification is that a customer might base their product on ATv2, and want P-GPS, but not have enough flash left due to their customer-specific code to allow for both the secondary mcuboot partition as well as a dedicated P-GPS partition.  Maybe they are also using memfault.

You are right, CLOUD_PGPS_MCUBOOT_PARTITION is a dummy symbol; it is simply there to have the `select BOOTLOADER_MCUBOOT` and `select IMG_MANAGER` side effect.

I could just remove this and let the customer figure it out.  What do you think?

Reply:
How about having a build instead assert? Then we can fail with some nice help text if if `BOOTLOADER_MCUBOOT` and `IMG_MANAGER` isn't selected for some reason.

Something like this:

`
#if enabled(CONFIG_NRF_CLOUD_PGPS) && !enabled(CONFIG_NRF_CLOUD_PGPS_PARTITION)
BUILD_ASSERT(IS_ENABLED(CONFIG_BOOTLOADER_MCUBOOT) && IS_ENABLED(CONFIG_IMG_MANAGER),
                         "Some help text")
#endif
 `

Reply:
I can do that.  However, I was picturing 3 ways to specify the location in flash to use:
1) enable CONFIG_NRF_CLOUD_PGPS_PARTITION
2) enable CONFIG_BOOTLOADER_MCUBOOT etc. and use PM_MCUBOOT_SECONDARY_ADDRESS and _SIZE
3) user uses a custom location from some other flash partition, or they create their own partition

Maybe that's making things too complicated.

Reply:
I see. But, wouldn't this config then basically auto select `CONFIG_BOOTLOADER_MCUBOOT ` and `CONFIG_IMG_MANAGER` in the third case as well? Maybe `depends on` is a lighter choice?

**Suggestion**

```
config CLOUD_PGPS_MCUBOOT_PARTITION
	bool "P-GPS MCUBoot partition"
	depends on !NRF_CLOUD_PGPS_PARTITION
	depends on BOOTLOADER_MCUBOOT
	depends on IMG_MANAGER
```

In code:

```
#if defined(CONFIG_CLOUD_PGPS_MCUBOOT_PARTITION)
			.storage_base = PM_MCUBOOT_SECONDARY_ADDRESS,
			.storage_size = PM_MCUBOOT_SECONDARY_SIZE
#
```

This approach also eliminates the need of guarding the `CLOUD_PGPS_MCUBOOT_PARTITION` with
`NRF_CLOUD_PGPS && !NRF_CLOUD_PGPS_PARTITION` and makes use of the kconfig in code.

Reply:
Actually, here's a better solution.  I've made a Kconfig choice with 3 possibilities.  This moves all the secondary MCUboot stuff to the P-GPS library if that choice is selected, and so the changes to samples and applications are less invasive.

```
choice NRF_CLOUD_PGPS_STORAGE
	prompt "nRF Cloud P-GPS persistent storage location"
	default NRF_CLOUD_PGPS_STORAGE_PARTITION if BUILD_S1_VARIANT
	help
	  The P-GPS subsystem requires a contiguous section of flash memory in
	  which to store predictions. The amount required is 2048 bytes times
	  the number of predictions set by NRF_CLOUD_PGPS_NUM_PREDICTIONS.
	  If mcuboot DFU is enabled, default to NRF_CLOUD_PGPS_STORAGE_PARTITION.
	  Otherwise, P-GPS will prevent a successful mcuboot update, causing it
	  to revert to the previous image.

config NRF_CLOUD_PGPS_STORAGE_PARTITION
	bool "Define P-GPS subsystem flash partition"
	select PM_SINGLE_IMAGE if !SPM
	help
	  If set, allocates flash for storing predictions.

config NRF_CLOUD_PGPS_STORAGE_MCUBOOT_SECONDARY
	bool "Reuse MCUboot secondary partition for P-GPS storage"
	select BOOTLOADER_MCUBOOT
	select IMG_MANAGER
       depends on !BUILD_S1_VARIANT
	help
	  If set, use the secondary MCUboot partition to also store P-GPS data.
	  Note that when a FOTA job executes, this data will be lost and will
	  have to be downloaded again after the FOTA job is complete. This is
	  not compatible with FOTA updates of MUCboot itself, which is possible
	  when SECURE_BOOT and BUILD_S1_VARIANT are enabled.

config NRF_CLOUD_PGPS_STORAGE_CUSTOM
	bool "Use application-provided location and size in flash"
	help
	  The application is responsible for providing the location in flash
	  in which to store P-GPS data.  The location must be aligned to a
	  flash page boundary (4096 bytes), and the size must be at least 2048
	  bytes times NRF_COUND_PGPS_NUM_PREDICTIONS.

endchoice # NRF_CLOUD_PGPS_STORAGE

Reply:
I could change the `select BOOTLOADER_MCUBOOT` and `select IMG_MANAGER` to `depends on` if you think that's better.

Reply:
I think `select` makes more sense with this approach. Looks good!